### PR TITLE
DSPDC-428 Add pure-ftpd chart

### PR DESCRIPTION
**Pull request by danxmoran**
This only supports one user for now. That's all we should need for testing, and we can always add support for > 1 user later.

### CODE REVIEW

**Comment by smark88** [comment]
Write PSP/RBAC helm charts 

[examples here](https://github.com/broadinstitute/datarepo-helm/tree/master/charts/datarepo-ui/templates)

File: charts/pure-ftpd/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            - name: USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .secretName }}
+                  key: {{ .usernameKey }}
+            - name: PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .secretName }}
+                  key: {{ .passwordKey }}
```
Comment (raaidbroad) [inline charts/pure-ftpd/templates/deployment.yaml:65]: My bad if I'm reading this wrong, but should all of these secrets refer to the same `.secretName` and `.passwordKey`?

Comment (smark88) [reply charts/pure-ftpd/templates/deployment.yaml:65]: The secret name contains the key and both could be any value.

**Comment by smark88** [comment]
Are you going to supply a default value.yaml or is the schema enough

File: charts/pure-ftpd/templates/deployment.yaml (hunk 1 above)
Comment (raaidbroad) [reply charts/pure-ftpd/templates/deployment.yaml:65]: Okay. If one of y'all could walk me through how that works at some point tomorrow, I'd v much appreciate it =)

**Review by raaidbroad — APPROVED** [review]

**Review by smark88 — CHANGES_REQUESTED** [review]
Missing [roleBinding](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)

[example] (https://github.com/broadinstitute/datarepo-helm/blob/master/charts/datarepo-ui/templates/rolebinding.yaml)

File: charts/pure-ftpd/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{{/* vim: set filetype=mustache: */}}
+
+{{/* Create the name of the service account to use */}}
+{{- define "pure-ftpd.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create -}}
+  {{ default (printf "%s-account" .Release.Name) .Values.serviceAccount.name }}
+{{- else -}}
+  {{ default "default" .Values.serviceAccount.name }}
+{{- end -}}
+{{- end -}}
```
Comment (smark88) [inline charts/pure-ftpd/templates/_helpers.tpl:24]: I would add the _helper.tpl for the labels here just for organization. 
```
Common labels
*/}}
{{- define "datarepo-ui.labels" -}}
helm.sh/chart: {{ include "datarepo-ui.chart" . }}
{{ include "datarepo-ui.selectorLabels" . }}
{{- if .Chart.AppVersion }}
app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
{{- end }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
{{- end -}}
```

File: charts/pure-ftpd/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
```
Comment (smark88) [inline charts/pure-ftpd/templates/deployment.yaml:3]: then add the common labels here ```
  labels:
    {{- include "datarepo-ui.labels" . | nindent 4 }}
```

**Comment by smark88** [comment]
Other than the labels it all looks good to me

**Pull request merged**